tests/kerastuner/applications/resnet_test.py

```diff
@@ -0,0 +1,58 @@
+
+# Copyright 2019 The Keras Tuner Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for HyperResnet Model."""
+
+import pytest
+
+from kerastuner.applications import resnet
+from kerastuner.engine import hyperparameters as hp_module
+
+
+def test_model_construction():
+    hp = hp_module.HyperParameters()
+    hypermodel = resnet.HyperResnet(input_shape=(256, 256, 3), num_classes=10)
+    model = hypermodel.build(hp)
```

Comment:
We should run basic tests on the model to check it is a valid model with the expected behavior (run it on a test input to check output shape; try train_on_batch to test compile config...)

For best coverage, we should do such checks for every possible configuration (albeit that may be challenging), otherwise we can't be sure there aren't invalid configurations in the mix.

Reply:
Done, added tests parametrized by the version, this test is now running in ~230 sec on my laptop so I wasn't sure if I should add more?

Reply:
Test are stalling unfortunately: https://travis-ci.org/keras-team/keras-tuner/jobs/547946699

Let's only run the top 5 smallest models on CI, and configure the test to run all models when run locally.